interview-questions.md

```diff
@@ -8,36 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: If you forget to add a foreign key to your Student model the foreign key to the Cohort model, you can generate a migration using the following command: ```rails generate migration AddForeignKeyToStudents cohort:references``` This will create a migration file that adds a foreign key to the students table, referencing the cohorts table. The foreign key name would be `cohort_id`, which is the default name generated by Rails when using the `references` attribute. The foreign key would be on the Student model, as it's the model that belongs to the Cohort model. So, the migration generated will add a cohort_id column to the students table
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The RESTful routes that must always have params passed are the POST and PUT/PATCH routes. The reason for the POST route requiring params is that you need params to be inputted when creating (POST) a new entity into the database. Similar to the POST route, the PUT/PATCH route require params because without them it isn't possible to locate or update an entity within the database.
 
-Researched answer:
+Researched answer: There are two RESTful routes that must always be passed params: POST, and PUT/PATCH. For the POST route, you must pass params to provide the necessary information to create the resource. Without these params, the resource cannot be created. Also, when updating an existing resource using the PUT/PATCH routes, you must pass params that specify what changes should be made to the resource. Without these params, the resource will not be updated.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: The names of some of the rails generator commands I use pretty often are: rails g migration, rails g controller, and rails g model.
 
-Researched answer:
+rails g migration: Generates a migration within the migrate directory that contains the migration name and attributes. Migrations are used to update/change the database in different ways. To execute the changes when you are done in the migrate directory you would run rails db:migrate in the console.
+
+rails g controller: Running this command in the terminal creates a new controller within the controllers directory. Optional Action methods can be added to the controller file.
+
+rails g model: Running this command create a model within the models directory and also generates a migration file in the migrate directory to create a data base. Attributes can be specified when running this command as well to create columns within the database. You can find the structure of the database within the schema.
+
+Researched answer: The name of three commonly used rails generator commands are: rails g controller, rails g migrate, rails g model:
```

Comment:
Excellent work on this question, I would use the researched answer to look into some other generate commands and talk about what they do such as resource and scaffold are pretty cool to talk about.